tiledb/sm/query/query_condition.cc

```diff
@@ -352,33 +350,77 @@ void QueryCondition::apply_clause(
       }
     } else {
       const auto tile_tuple = result_tile->tile_tuple(field_name);
-      const auto& tile = std::get<0>(*tile_tuple);
-      const uint64_t cell_size = tile.cell_size();
-      ChunkedBuffer* const chunked_buffer = tile.chunked_buffer();
-      char* const buffer =
-          static_cast<char*>(chunked_buffer->get_contiguous_unsafe());
+      char* buffer = nullptr;
+      uint64_t buffer_size = 0;
+      uint64_t* buffer_offsets = nullptr;
+      uint64_t buffer_offsets_el = 0;
+      uint8_t* buffer_validity = nullptr;
+      uint64_t cell_size = 0;
+      uint64_t buffer_offset = 0;
+      uint64_t buffer_offset_inc = 0;
+
+      if (var_size) {
+        const auto& tile = std::get<1>(*tile_tuple);
+        buffer =
+            static_cast<char*>(tile.chunked_buffer()->get_contiguous_unsafe());
+        buffer_size = tile.size();
+
+        const auto& tile_offsets = std::get<0>(*tile_tuple);
+        buffer_offsets = static_cast<uint64_t*>(
+            tile_offsets.chunked_buffer()->get_contiguous_unsafe());
+        buffer_offsets_el =
+            tile_offsets.size() / constants::cell_var_offset_size;
+      } else {
+        const auto& tile = std::get<0>(*tile_tuple);
+        buffer =
+            static_cast<char*>(tile.chunked_buffer()->get_contiguous_unsafe());
+        cell_size = tile.cell_size();
+        buffer_offset = start * cell_size;
+        buffer_offset_inc = stride * cell_size;
+      }
+
+      if (nullable) {
+        const auto& tile_validity = std::get<2>(*tile_tuple);
+        buffer_validity = static_cast<uint8_t*>(
+            tile_validity.chunked_buffer()->get_contiguous_unsafe());
+      }
 
       // Start the pending result cell slab at the start position
       // of the current result cell slab.
       uint64_t pending_start = start;
 
       // Iterate through each cell in this slab.
       uint64_t c = 0;
-      uint64_t buffer_offset = start * cell_size;
-      uint64_t buffer_offset_inc = stride * cell_size;
+      bool null_cell = false;
       while (c < length) {
+        if (var_size) {
```

Comment:
It may be worth moving the if condition outside the loop for each of the cases.

Reply:
I tried doing so but ended up with lots of duplicated code... Thoughts?

Reply:
Perhaps that's fine for now as I expect this path to be refactored in the very near future. Duplication is necessary for performance (we need to get rid of all the branching statements in the loops), but we can punt.

Reply:
Fixed :)